### DISCUSSION TITLE
SQlite3: issue a `PRAGMA optimise` before closing each database connection

### DISCUSSION THREAD

**Discussion by solracsf** [juicedata/juicefs#2806] [General]
**What would you like to be added**:

To achieve the best long-term query performance without the need to do a detailed engineering analysis of the application schema and SQL, it is recommended that applications run "`PRAGMA optimize`" (with no arguments) just before closing each [database connection](https://www.sqlite.org/c3ref/sqlite3.html).

This pragma is usually a no-op or nearly so and is very fast.

**Why is this needed**:

https://www.sqlite.org/pragma.html#pragma_optimize

**Comment by davies** [comment]
It's better to do this outside of juicefs, for example, before mount or after umount.

**Reply by solracsf** [reply]
How can this be done on mount/umount and not application (juicefs) side as it is recommended to be performed
>"just before closing **each database connection**"

?

**Reply by davies** [reply]
It's used to optimize the database in disk, can be issued by `sqlite cli`.
